cluster/permissions.go

```diff
@@ -902,3 +908,60 @@ func MapPermissionsToIDs(ctx *Context, permissions []string) (map[string]*uuid.U
 	return permToID, nil
 
 }
+
+var ErrDuplicatedPermission = errors.New("Another permission for those actions, effect and resources already exists")
+
+func validatePermissionUniqueness(ctx *Context, effect Effect, resources, actions []string) error {
+	// we are going to query for matching permissions, if the following query returns at least 1 result, there's another
+	// permission with these capabilities (effect-actions-resources)
+
+	var buckets []string
+	var completeResources []string
+	for _, res := range resources {
+		parts := strings.Split(res, "/")
+		r := Resource{}
+		if len(parts) > 0 {
+			buckets = append(buckets, parts[0])
+			r.BucketName = parts[0]
+		}
+		if len(parts) > 1 {
+			r.Pattern = parts[1]
+		} else {
+			r.Pattern = "*"
+		}
+		completeResources = append(completeResources, r.String())
+	}
+
+	query := `SELECT p.id
```

Comment:
Just to detect if the permissions are unique why do you need to call the database @dvaldivia ? or am I missing something here?

Reply:
We keep track of the permissions on the DB so as we are adding a new one, we validate via query.

Reply:
I think the issue is when assigning permissions with the same rules to  a Service Account, not the creation of a permission itself. I'm gonna do some tests with this pr.

Reply:
yes, I thinks this doesn't fix the real issue which is there is the same effect on a certain bucket in two different permissions.
<img width="313" alt="Screen Shot 2020-01-15 at 5 42 04 PM" src="https://user-images.githubusercontent.com/11819101/72486015-784f5380-37be-11ea-8edd-c98a44f60c53.png">
 and gives :
 Error assigning permissions to service account: MinIO: `policy.MarshalJSON`, duplicate actions [s3:*], resources [arn:aws:s3:::bucket1/*] found in statements { Allow [s3:*] [arn:aws:s3:::bucket1/*] []}, { Allow [s3:*] [arn:aws:s3:::bucket1/*] []}

Reply:
fixed